stl/inc/algorithm

```diff
@@ -1564,28 +1564,36 @@ namespace ranges {
         constexpr copy_if_result<_It, _Out> operator()(
             _It _First, _Se _Last, _Out _Result, _Pr _Pred, _Pj _Proj = {}) const {
             _Adl_verify_range(_First, _Last);
-            auto _UFirst      = _Get_unwrapped(_STD move(_First));
-            const auto _ULast = _Get_unwrapped(_STD move(_Last));
-            for (; _UFirst != _ULast; ++_UFirst) {
-                if (_STD invoke(_Pred, _STD invoke(_Proj, *_UFirst))) {
-                    *_Result = *_UFirst;
-                    ++_Result;
-                }
-            }
-
-            _Seek_wrapped(_First, _STD move(_UFirst));
-            return {_STD move(_First), _STD move(_Result)};
+            auto _Res = _Copy_if_unchecked(_Get_unwrapped(_STD move(_First)), _Get_unwrapped(_STD move(_Last)),
+                                           _STD move(_Result), _Pass_fn(_Pred), _Pass_fn(_Proj));
+            _Seek_wrapped(_First, _STD move(_Res.in));
+            return {_STD move(_First), _STD move(_Res.out)};
         }
 
         template <input_range _Rng, weakly_incrementable _Out, class _Pj = identity,
             indirect_unary_predicate<projected<iterator_t<_Rng>, _Pj>> _Pr>
             requires indirectly_copyable<iterator_t<_Rng>, _Out>
         constexpr copy_if_result<borrowed_iterator_t<_Rng>, _Out> operator()(
             _Rng&& _Range, _Out _Result, _Pr _Pred, _Pj _Proj = {}) const {
-            return (*this)(
-                _RANGES begin(_Range), _RANGES end(_Range), _STD move(_Result), _Pass_fn(_Pred), _Pass_fn(_Proj));
+            auto _Res = _Copy_if_unchecked(_Ubegin(_Range), _Uend(_Range),
+                                           _STD move(_Result), _Pass_fn(_Pred), _Pass_fn(_Proj));
+            auto _First = _RANGES begin(_Range);
+            _Seek_wrapped(_First, _STD move(_Res.in));
+            return {_STD move(_First), _STD move(_Res.out)};
         }
         // clang-format on
+    private:
+        template <class _It, class _Se, class _Out, class _Pj, class _Pr>
+        _NODISCARD constexpr copy_if_result<_It, _Out> _Copy_if_unchecked(

```

Comment:
```suggestion
        _NODISCARD static constexpr copy_if_result<_It, _Out> _Copy_if_unchecked(
```

Reply:
Will change but remove the const

Reply:
Done

Reply:
...and now I'm wondering if I missed this in the other reviews.